dipy/nn/cnn_1denoiser/cnn_1denoiser.py

```diff
@@ -0,0 +1,283 @@
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.layers import Conv1D, Activation
+from tensorflow.keras.models import *
+from tensorflow.keras.layers import Input
+#from patchify import patchify, unpatchify
+from sklearn.model_selection import train_test_split
+from datetime import datetime
+now = datetime.utcnow().strftime("%Y%m%d%H%M%S")
+root_logdir = "tf_logs"
+logdir = "{}/run-{}/".format(root_logdir, now)
+'''
+Title : Denoising diffusion weighted imaging data using CNN
+=====
+
+we constructed a simple 1D-CNN model that has five layers, including two convolutional layers, each
+followed by a max-pooling layer, and a dense layer (Figure 1). The first convolutional layer was given an
+input of the ‘noisy’ high-noise image and consisted of 16 one-dimensional filtering kernels of size 16. The
+second convolutional layer consisted of 32 one-dimensional filtering kernels of size 8. The ReLu
+activation function was used in both convolutional layers. There was 1 max-pooling layer that had a kernel
+size of 2 with stride 2. In the dense layer, the extracted features of the high-noise image were mapped to
+the low-noise reference image.
+
+reference:
+---------
+Denoising diffusion weighted imaging data using convolutional neural networks
+Hu Cheng, Sophia Vinci-Booher, Jian Wang, Bradley Caron, Qiuting Wen, Sharlene Newman, Franco Pestilli
+bioRxiv 2022.01.17.476708; doi: https://doi.org/10.1101/2022.01.17.476708
+
+Contributors:
+------------
+-Rahul Suresh Ubale :implemented the cnn_1denoiser model and tested the code.
+-Hu Cheng :provided guidance and support for the project.
+-Eleftherios Garyfallidis :provided guidance and support for the project.
+'''
+class cnn_1denoiser:
```

Comment:
I am not a fan of this name. It's also not PEP8 compliant as a class name. How about "Cnn1DDenoiser"?

Reply:
I'd also change the module name to "cnn_1d_denoising"

Reply:
Done